### PR TITLE
correct pluralization for vertex, human

### LINKED ISSUE #11515: Pluralize issue: 'vertex' -> 'vertexes'
**Do you want to request a *feature* or report a *bug*?**
BUG

**What is the current behavior?**
pluralize('vertex') === 'vertexes'

**If the current behavior is a bug, please provide the steps to reproduce.**

**What is the expected behavior?**
pluralize('vertex') === 'vertices'

**What are the versions of Node.js, Mongoose and MongoDB you are using? Note that "latest" is not a version.**

### LINKED ISSUE #13037: Passing "Human" as model name creates the collection as "Humen"
### Prerequisites

- [X] I have searched existing issues to ensure the bug has not already been reported

- [X] I have written a descriptive issue title

### Mongoose version

6.9.1

### Node.js version

18.13.0

### MongoDB server version

6.0.4

### Typescript version (if applicable)

_No response_

### Description

When pluralizing the collection name, mongoose sets it to "Humen". Plural for human is not humen, it's not a real word. The correct plural form is "Humans".

### Steps to Reproduce

Simply naming passing "Human" to the model function results in this bug. (`const Human = mongoose.model("Human", humanSchema);`). 

### Expected Behavior

Should be set to "Humans" instead.

### PR DESCRIPTION
**Pull request by vkarpov15**
Fix Automattic/mongoose#11515
Fix Automattic/mongoose#13037

@hasezoey @uzlopak can you please take a look?

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -4,8 +4,6 @@ module.exports = pluralize;
  * Pluralization rules.
  *
  * These rules are applied while processing the argument to `toCollectionName`.
- *
- * @deprecated remove in 4.x gh-1350
  */
 
 exports.pluralization = [
```
Comment (Uzlopak) [inline index.js:9]:
```suggestion
exports.pluralization = [
  [/human$/gi, '$1ens'],
```

**Comment by Uzlopak** [comment]
The line `[/(x|ch|ss|sh)$/gi, '$1es'],` has to be switched with the following line handling vertrex. Than you dont need the Map with the exceptions

**Review by hasezoey — APPROVED** [review]
LGTM

**Comment by vkarpov15** [comment]
On second thought, I think we should leave the plural of "vertex" as "vertexes" to be consistent with the discussion from this issue: https://github.com/Automattic/mongoose/pull/11953 . TLDR; we argued that "indexes" is an acceptable plural form of "index" because major dictionaries like Oxford and Merriam-Webster agree. "vertexes" is sufficiently similar.

**Comment by vkarpov15** [comment]
Fixed handling for 'human' and related pluralizations, e.g. 'superhuman' -> 'superhumans', etc.

```
$ node
Welcome to Node.js v16.17.0.
Type ".help" for more information.
> pluralize('human')
Uncaught ReferenceError: pluralize is not defined
> const pluralize = require('./')
undefined
> pluralize('human')
'humans'
> pluralize('superhuman')
'superhumans'
> pluralize('man')
'men'
> pluralize('ombudsman')
'ombudsmen'
> 
(To exit, press Ctrl+C again or Ctrl+D or type .exit)
> 

```

**Pull request merged**